Wikimate.php

```diff
@@ -1306,6 +1318,157 @@ public function getWidth()
 
 	/*
 	 *
+	 * File history & deletion methods
+	 *
+	 */
+
+	/**
+	 * Returns the revision history of this file with all properties.
+	 * The initial history at object creation contains only the
+	 * current revision of the file. To obtain more revisions,
+	 * set $refresh to true and also optionally set $limit and
+	 * the timestamps.
+	 *
+	 * The maximum limit is 500 for user accounts and 5000 for bot accounts.
+	 *
+	 * Timestamps can be in several formats as described here:
+	 * https://www.mediawiki.org/w/api.php?action=help&modules=main#main.2Fdatatypes
+	 *
+	 * @param   boolean  $refresh  True to query the wiki API again
+	 * @param   integer  $limit    The number of file revisions to return
+	 *                             (the maximum number by default)
+	 * @param   string   $startts  The start timestamp of the listing (optional)
+	 * @param   string   $endts    The end timestamp of the listing (optional)
+	 * @return  mixed              The array of selected file revisions, or null if error
+	 */
+	public function getHistory($refresh = false, $limit = null, $startts = null, $endts = null)
+	{
+		if ($refresh) { // We want to query the API
+			// Collect optional history parameters
+			$history = array();
+			if (!is_null($limit)) {
+				$history['iilimit'] = $limit;
+			} else {
+				$history['iilimit'] = 'max';
+			}
+			if (!is_null($startts)) {
+				$history['iistart'] = $startts;
+			}
+			if (!is_null($endts)) {
+				$history['iiend'] = $endts;
+			}
+
+			// Get file revision history
+			if ($this->getInfo($refresh, $history) === null) {
+				return null;
+			}
+		}
+
+		return $this->history;
+	}
+
+	/**
+	 * Returns the properties of the specified file revision.
+	 *
+	 * Revision can be the following:
+	 * - revision timestamp (string:"2001-01-15T14:56:00Z")
+	 * - revision index (int:3)
+	 * The current revision has index 0.
+	 *
+	 * @param   mixed  $revision  The index or timestamp of the revision
+	 * @return  mixed             The properties (array), or null if not found
+	 */
+	public function getRevision($revision)
+	{
+		// Select revision by index
+		if (is_int($revision)) {
+			if (isset($this->history[$revision])) {
+				return $this->history[$revision];
+			}
+		// Search revision by timestamp
+		} else {
+			foreach ($this->history as $history) {
+				if ($history['timestamp'] == $revision) {
+					return $history;
+				}
+			}
+		}
+
+		// Return error message
+		$this->error = array();
+		$this->error['file'] = 'The revision is not found for this file';
```

Comment:
I'd mention the actual revision parameter passed in the error message, and also change "is not found" --> "was not found".